sdk/messaging/azeventhubs/checkpointstore/blob_checkpoint_store.go

```diff
@@ -0,0 +1,398 @@
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &blobCheckpointStore{
+		cc: cc,
+	}, nil
+}
+
+// NewBlobCheckpointStore creates a checkpoint store that stores ownership and checkpoints in
+// Azure Blob storage. The container client will be used to store blobs.
+// NOTE: the container must exist before the checkpoint store can be used.
+func NewFromConnectionString(connectionString string, containerName string, options *NewOptions) (azeventhubs.CheckpointStore, error) {
+	cc, err := blob.NewContainerClientFromConnectionString(connectionString, containerName, toContainerClientOptions(options))
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &blobCheckpointStore{
+		cc: cc,
+	}, nil
+}
+
+// NewBlobCheckpointStore creates a checkpoint store that stores ownership and checkpoints in
+// Azure Blob storage. The container client will be used to store blobs.
+// NOTE: the container must exist before the checkpoint store can be used.
+func NewWithSharedKey(containerURL string, cred *SharedKeyCredential, options *NewOptions) (azeventhubs.CheckpointStore, error) {
+	cc, err := blob.NewContainerClientWithSharedKey(containerURL, cred.key, toContainerClientOptions(options))
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &blobCheckpointStore{
+		cc: cc,
+	}, nil
+}
+
+type blobCheckpointStore struct {
+	cc *blob.ContainerClient
+}
+
+func (b *blobCheckpointStore) ClaimOwnership(ctx context.Context, partitionOwnership []azeventhubs.Ownership, options *azeventhubs.ClaimOwnershipOptions) ([]azeventhubs.Ownership, error) {
+	var ownerships []azeventhubs.Ownership
+
+	// TODO: in parallel?
+	for _, po := range partitionOwnership {
+		blobName, err := nameForOwnershipBlob(po.CheckpointStoreAddress)
+
+		if err != nil {
+			return nil, err
+		}
+
+		var expectedETag *string
+
+		if po.ETag != "" {
+			expectedETag = &po.ETag
+		}
+
+		lastModified, etag, err := b.setMetadata(ctx, blobName, newOwnershipBlobMetadata(po.OwnershipData), expectedETag)
+
+		if err != nil {
+			var storageErr *blob.StorageError
+
+			// we can fail to claim ownership and that's okay - it's expected that clients will
+			// attempt to claim with whatever state they hold locally. If they fail it just means
+			// someone else claimed ownership before them.
+			if errors.As(err, &storageErr) && storageErr.ErrorCode == blob.StorageErrorCodeConditionNotMet {
+				continue
+			}
+
+			return nil, err
+		}
+
+		ownerships = append(ownerships, azeventhubs.Ownership{
+			CheckpointStoreAddress: po.CheckpointStoreAddress,
+			OwnershipData: azeventhubs.OwnershipData{
+				ETag:             etag,
+				OwnerID:          po.OwnershipData.OwnerID,
+				LastModifiedTime: *lastModified,
+			},
+		})
+	}
+
+	return ownerships, nil
+}
+
+func (b *blobCheckpointStore) ListCheckpoints(ctx context.Context, fullyQualifiedNamespace string, eventHubName string, consumerGroup string, options *azeventhubs.ListCheckpointsOptions) ([]azeventhubs.Checkpoint, error) {
```

Comment:
Should this return a *runtime.Pager[T] instead? How many ownerships are typical?

Reply:
32, max. So probably not worth a pager.

Reply:
With that said, one change @jsquire  and I have been talking about is just getting the checkpoint by name, rather than listing them all, as that's the more common case.